core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -453,7 +454,7 @@ Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
     /**
      * Begin merge query
      */
-    MergeHandle beginMerge(Session session, TableHandle tableHandle);
+    MergeHandle beginMerge(Session session, TableHandle tableHandle, Multimap<Integer, ColumnHandle> updateCaseColumnHandles);
```

Comment:
What is meaning of the key? I think this could be explained in javadoc.

Reply:
I leave the `The merge update case number with the relevant columns mapping`, but still want to get your suggestions, could you please help to phrase the sentence?  cc @kokosing @djsstarburst 
The mapping here is the `MergeUpdate` case number to it's setExpressions assignment target, then update would known what are the target columns. Take the a example:
```
MERGE INTO t USING s 
ON t.pk = s.pk 
WHEN MATCHED AND t.c1 > 0 THEN DELETE // case 0
WHEN MATCHED AND t.c2 < 10 THEN UPDATE SET t.c2 = 10, t.c3 = 10 // case 1
WHEN MATCHED AND t.c2 < 100 THEN UPDATE SET t.c4 = 100, t.c3 = 100 // case 2
WHEN MATCHED THEN UPDATE SET t.c5 = s.c5 + 10 // case 3
WHEN NOT MATCHED THEN INSERT.... // case 4
``` 
the map value would be like: {1: [c2, c3], 2: [c4, c3], 3: c5}